src/clj_telnet/core.clj

```diff
@@ -69,15 +74,11 @@
 
 (defn write
   "writes to the output stream of a telnet client"
-  [^TelnetClient telnet ^String s]
-  (let [out (PrintStream. (.getOutputStream telnet))]
-    (doto out
-      (.println s)
-      (.flush))))
-
-(defn with-telnet
-  [telnet f]
-  (try (f telnet)
-       (catch Exception e
-         (kill-telnet telnet)
-         (throw e))))
+  ([^TelnetClient telnet ^String s cr]
```

Comment:
This is completely changes the behavior of with-telnet.
I'm not a fan.

Reply:
It should probably be a macro that cleans up and throws the exception.

Reply:
https://github.com/komcrad/clj-telnet/pull/15/commits/26ddcae530c4ba1e894df6935ce2f8780ad2c393

This was a good commit. It makes with-telnet into what I would want.
I wanted something that would throw the exception. So adding with-open compatibility and re-writing with-telnet were reasonable changes.